### PR TITLE
5526 - Hide empty partitions in Inbox

### PR DESCRIPTION
**Pull request by robbiecarlton**

### CODE REVIEW

**Comment by robbiecarlton** [comment]
See comment with skipped test. @JettTech I spent some more time working on that test this morning with no luck and eventually agreed with @SirRobLyon that it was not worth more time to write a test for a cosmetic change that works in the app.

File: src/holofuel/pages/Inbox/Inbox.js [hunk 1]
```diff
@@ -258,27 +267,25 @@ export function TransactionRow ({ transaction, setActionsVisibleId, actionsVisib
...
 
   const onConfirmGreen = () => {
     setHighlightGreen(true)
     setIsDisabled(true)
     setTimeout(() => {
       setHighlightGreen(false)
-      setIsVisible(false)
-    }, 5000)
+      hideTransaction()
+    }, 1000)
```
Comment (JettTech) [inline src/holofuel/pages/Inbox/Inbox.js]: Can we return this to 5000ms?

File: src/holofuel/pages/Inbox/Inbox.js [hunk 2]
```diff
@@ -336,13 +343,11 @@ export function TransactionRow ({ transaction, setActionsVisibleId, actionsVisib
 
     {isActionable && !isLoading && !isDisabled && <>
       <RevealActionsButton
-        actionsVisibleId={actionsVisibleId}
-        visible={drawerIsOpen}
-        actionsClick={() => setActionsVisibleId(transaction.id)}
-        handleClose={handleCloseReveal}
+        isDrawerOpen={isDrawerOpen}
+        openDrawer={() => setIsDrawerOpen(true)}
+        closeDrawer={() => setIsDrawerOpen(false)}
```
Comment (JettTech) [inline src/holofuel/pages/Inbox/Inbox.js:360]: Replacing the knowledge of which transaction is open with a simple boolean, will allow multiple transactions to be open at once.  Even though this boolean is simpler to read, I do believe we need to restore this state knowledge.  (We had explicitly prevented the multiple drawer opening behavior during our UX iterations with Jamison last fall.)

File: src/holofuel/pages/Inbox/Inbox.test.js [hunk 3]
```diff
@@ -19,6 +19,7 @@ import HolofuelDeclineMutation from 'graphql/HolofuelDeclineMutation.gql'
 import HolofuelCounterpartyQuery from 'graphql/HolofuelCounterpartyQuery.gql'
 import HolofuelUserQuery from 'graphql/HolofuelUserQuery.gql'
 import { presentAgentId, promiseMap } from '../../../utils'
+import { DIRECTION } from '../../../models/Transaction'
```
Comment (JettTech) [inline src/holofuel/pages/Inbox/Inbox.test.js]: This is a bit nit-picky, but let's replace the relative path with the absolute (from src), just to keep things clean and consistent.

File: src/holofuel/pages/Inbox/Inbox.test.js [hunk 4]
```diff
@@ -169,6 +170,149 @@ const whoamiMock = {
...
+      wait(1000)
+    })
+
+    await act(async () => {
+      fireEvent.click(getByText('Accept'))
+    })
+
+    await act(async () => {
+      fireEvent.click(getByText('Yes'))
+      jest.runAllTimers()
```
Comment (JettTech) [inline src/holofuel/pages/Inbox/Inbox.test.js:296]: :1st_place_medal: Yea!! Glad we're using this - this is gold!

File: src/pages/Dashboard/Dashboard.js [hunk 5]
```diff
@@ -20,8 +19,8 @@ export const mockEarnings = 4984
 export default function Dashboard ({ earnings = mockEarnings }) {
   const { data: { hposSettings: settings = [] } = {} } = useQuery(HposSettingsQuery)
 
-  const { data: { happs = [] } = {} } = useQuery(HappsQuery)
-  const noInstalledHapps = happs.reduce((total, happ) => happ.isEnabled ? total + 1 : total, 0)
+  // placeholder as we're not currently calling hha
+  const noInstalledHapps = 0
```
Comment (JettTech) [inline src/pages/Dashboard/Dashboard.js:23]: :+1:

File: src/holofuel/pages/Inbox/Inbox.test.js [hunk 6]
```diff
@@ -169,6 +170,149 @@ const whoamiMock = {
...
+      ledgerMock,
+      actionableTransactionsMock,
+      actionableTransactionsMock,
+      actionableTransactionsMock,
+      acceptOffer1Mock,
+      acceptOffer2Mock,
+      counterpartyMock,
+      counterpartyMock,
+      counterpartyMock
+    ]
```
Comment (JettTech) [inline src/holofuel/pages/Inbox/Inbox.test.js:261]: Just to capture our conversation with this update, I'll write the overview decision about this testing pattern here.

> We decided that this approach is not optimal, but that we would keep this approach for now.  We are interested in finding a better testing design that would prevent multiple iterations of the same mock.

File: src/holofuel/pages/Inbox/Inbox.test.js [hunk 7]
```diff
@@ -169,6 +170,149 @@ const whoamiMock = {
...
+      data: {
+        holofuelAcceptOffer: offer2
+      }
+    }))
+  }
+
+  it.skip('hides a partition when there are no more transactions in that partition', async () => {
+    // This test is currently broken. The second time setTimeout is called in onConfirmGreen it doesn't run so we the
+    // app never gets to the final state. I'm skipping this test because I don't want to spend any more time on a test for a cosmetic
+    // feature that works
```
Comment (JettTech) [inline src/holofuel/pages/Inbox/Inbox.test.js:247]: Is there a ticket in PT to capture this?

File: src/holofuel/pages/Inbox/Inbox.js [hunk 8]
```diff
@@ -213,16 +203,35 @@ export default function Inbox ({ history: { push } }) {
   </PrimaryLayout>
 }
 
-export function TransactionRow ({ transaction, setActionsVisibleId, actionsVisibleId, setConfirmationModalProperties, isActionable, myProfile }) {
-  const { counterparty, amount, type, status, direction, notes, canceledBy, isPayingARequest } = transaction // presentBalance,
-  const agent = canceledBy || counterparty
+export function Partition ({ dateLabel, transactions, userId, setConfirmationModalProperties, isActionable }) {
+  const [hiddenTransactionIds, setHiddenTransactionIds] = useState([])
```
Comment (JettTech) [inline src/holofuel/pages/Inbox/Inbox.js]: What is the benefit of adding this additional layer of component abstraction?

**Review by JettTech — CHANGES_REQUESTED** [review]
A few small updates and a question.  Let me know if you want to chat.

File: src/holofuel/pages/Inbox/Inbox.js (hunk 8 above)
Comment (robbiecarlton) [reply src/holofuel/pages/Inbox/Inbox.js]: answered in person

**Comment by JettTech** [comment]
@robbiecarlton - Tests run and all looks well.  

Before merging, however, I would like to demo the change to the drawer UX with Rob and C to ensure that this matches their approval too.  If they approve the UX change, then I'm fine with the merge.

**Review by JettTech — APPROVED** [review]
LGTM!

**Pull request merged**